iotilebuild/iotile/build/config/site_scons/autobuild.py

```diff
@@ -211,3 +211,40 @@ def autobuild_documentation(tile):
     env.Command(outfile, docfile, action=env.Action(action, "Building Component Documentation"))
     Alias('documentation', outdir)
     env.Clean(outfile, outdir)
+
+def autobuild_bootstrap_file(file_name, image_list):
+    """Combine multiple firmware images into a single bootstrap hex file.
+    
+    Args:
+        file_name(str): Full name of output bootstrap hex file
+        image_list(list): List of files that will be combined into a single hex file
+            that will be used to flash a chip.
+    """
+    outputbase = os.path.join('build', 'output')
+    env = Environment(tools=[])
+    if platform.system() == 'Windows':
+        env = Environment(tools=['mingw'], ENV=os.environ)
+    else:
+        env = Environment(tools=['default'], ENV=os.environ)
+    
+    full_output_name      = os.path.join(outputbase,file_name)
+    full_image_list_names = []
+
+    for image in image_list:
+        root, ext = os.path.splitext(image)
+        if len(ext) == 0:
+            raise ArgumentError("Unknown file format or missing file extension", file_name=image)
+        file_format = ext[1:]
+        file = os.path.join(outputbase, image)
+
+        if file_format == 'hex':
+            full_image_list_names.append(file)
+        elif file_format == 'elf':
+            new_file = file.replace('.elf','.hex')
+            outhex = env.Command(new_file, file, "arm-none-eabi-objcopy -O ihex $SOURCE $TARGET")
+            full_image_list_names.append(new_file)
+            Delete(new_file)
```

Comment:
This line won't do what you expect it to because it isn't attached to a command that has a source and target so SCons doesn't know when to run it.  What you should do is keep track of all of the `new_files` that you need to make here and append them to the list of actions in the final `env.Comand` that you're creating at the end of the function as in:
http://scons.org/doc/1.2.0/HTML/scons-user/x2940.html

You'll have to build the list of actions step by step and then pass the list as a whole to `env.Command` instead of passing `action=`.  That way these temporary files will automatically be deleted as part of the build process after merging them.  You'll need to modify the test code to not reference the .hex files as well.  This will prevent polluting `build/output` with temporary `.hex` files.